### PR TITLE
Publish NuGet Package Microsot.FluentUI.ReactNative to internal Office NuGet Feed

### PR DESCRIPTION
**Pull request by Saadnajmi**
### Platforms Impacted
- [ ] iOS
- [ ] macOS
- [ ] win32 (Office)
- [ ] windows
- [ ] android

### Description of changes

We would like to make it easier to use FluentUI React Native in our internal code base. As such, we need to publish a NuGet package to make it consumable. This PR adds the appropriate nuspec and yml file changes to do so.

I added a template yml file with the required steps to publish to the Office NuGet feed. I also added my own Personal Access Token so that our Azure pipeline has access to that feed and publishing rights. The NuGet Package by default packs every file in the repo after it is built (including hidden .git/ files). 

### Verification

I tested out the changes in a separate pipeline with my fork of the repo. To make sure I didn't mess up the npm packages, I took care to only run the yml steps related to the nuget package, mostly by either commenting out the other steps or running only the publish-nuget-job.yml template You can see the pipeline here https://dev.azure.com/ms/ui-fabric-react-native/_build?definitionId=407

### Pull request checklist

This PR has considered (when applicable):
- [x] Automated Tests
- [ ] Documentation and examples
- [ ] Keyboard Accessibility
- [ ] Voiceover
- [ ] Internationalization and Right-to-left Layouts

### CODE REVIEW

File: scripts/generate_build_number.sh [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+fi
+
+if [ -n "$BUILD_BUILDNUMBER" ]; then
+    BUILD_VERSION_STRING="$BUILD_BUILDNUMBER"
+fi
+
+echo "Original Build Number: $BUILD_BUILDNUMBER"
+echo "Adjusted Build Number: $BUILD_VERSION_STRING$BUILD_VERSION_SUFFIX"
+
+echo "##vso[task.setvariable variable=sanitizedBuildNumber]$BUILD_VERSION_STRING$BUILD_VERSION_SUFFIX"
```
Comment (Saadnajmi) [inline scripts/generate_build_number.sh:24]: This file is copied from FluentUI Apple https://github.com/microsoft/fluentui-apple/blob/master/scripts/generate_build_number.sh

File: .vscode/settings.json [hunk 2]
```diff
@@ -61,5 +61,9 @@
   },
   "[handlebars]": {
     "editor.formatOnSave": false
-  }
+  },
+  "cSpell.words": [
+    "beachball",
+    "nuget"
+  ]
```
Comment (Saadnajmi) [inline .vscode/settings.json:68]: It seems these were added when I clicked "add word to workspace dictionary" while editing in vscode. Seems fine to include?

**Review by harrieshin — APPROVED** [review]

**Review by anandrajeswaran — APPROVED** [review]

**Review by HeyImChris — APPROVED** [review]

File: scripts/generate_build_number.sh (hunk 1 above)
Comment (Saadnajmi) [reply scripts/generate_build_number.sh:24]: Not sure what you mean? This build number will be unique though, yes

File: .ado/templates/publish-nuget-job.yml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  # generate the sanitizedBuildNumber environment variable
+  - bash: 'scripts/generate_build_number.sh'
+    displayName: 'Generate Build Number'
+
+  # pack the nuspec
+  - task: NuGetCommand@2
+    displayName: 'NuGet pack'
+    inputs:
+      command: pack
+      packagesToPack: 'FluentUI.nuspec'
```
Comment (anandrajeswaran) [inline .ado/templates/publish-nuget-job.yml:11]: This should probably be FluentUI.ReactNative.nuspec or something else to distinguish from other FluentUI?

File: scripts/generate_build_number.sh (hunk 1 above)
Comment (anandrajeswaran) [reply scripts/generate_build_number.sh:24]: Basically I was wondering if all platforms are aligned on this build number strategy or we're planning this to be Apple specific. Basically is this file run in publish for all platforms or just Apple?

Comment (Saadnajmi) [reply scripts/generate_build_number.sh:24]: Not planning this to be apple specific. I do wonder however if we want to prefix with the @fluentui/react-native semantic package version. Opened #480 to track

File: .ado/templates/publish-nuget-job.yml (hunk 3 above)
Comment (Saadnajmi) [reply .ado/templates/publish-nuget-job.yml:11]: the name doesn't actually matter. The id inside the nuspec matters, which is Microsoft.FluentUI.ReactNative

I'll probably rename this package.nuspec just to be more generic

**Pull request merged**